examples/ble-gatt-server/src/androidTest/java/no/nordicsemi/android/ble/ble_gatt_server/ExampleInstrumentedTest.kt

```diff
@@ -0,0 +1,24 @@
+package no.nordicsemi.android.ble.ble_gatt_server
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.ext.junit.runners.AndroidJUnit4
+
+import org.junit.Test
+import org.junit.runner.RunWith
+
+import org.junit.Assert.*
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
+    @Test
+    fun useAppContext() {
+        // Context of the app under test.
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        assertEquals("no.nordicsemi.android.ble.ble_gatt_server", appContext.packageName)
+    }
+}
```

Comment:
I'm thinking here of an on-device test for exercising bluetooth connectivity, but it may be a bit tricky, and certainly not applicable for the simulator.

Reply:
For iOS I did this lib: https://github.com/NordicSemiconductor/IOS-CoreBluetooth-Mock
where the whole `CBPeripheral` and `CBCentralManager` (equivalents of `BluetoothGatt` and `BluetoothAdapter`) are mocked and based on the flag a native or mock impl is chosen. Similar would be very good to have on Android, but... I imagine it should throw 133 at random moments and disconnects in other to fully "emulate the real behavior" ;)
Until then it's hard to test the lib and apps, so I mostly rely on my generous users who report bugs, except testing on Android during development. Not ideal, of course, but the best I can have at the moment.